test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,644 @@
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.test.common.AssertExtensions;
+import io.pravega.test.common.TestUtils;
+import io.pravega.test.common.TestingServerStarter;
+import io.pravega.test.integration.demo.ControllerWrapper;
+
+import lombok.SneakyThrows;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.curator.test.TestingServer;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.Random;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.ExecutionException;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Integration test for {@link KeyValueTable}s along with {@link Stream}s to check proper functioning
+ * of both primitives when used together using real Segment Store, Controller and connection.
+ */
+@Slf4j
+public class KVTableWithStreamTest extends KeyValueTableTestBase {
+
+    private static final String ENDPOINT = "localhost";
+    private static final String SCOPE = "Scope";
+    private static final KeyValueTableConfiguration DEFAULT_CONFIG = KeyValueTableConfiguration.builder()
+            .partitionCount(5).build();
+    private static final Duration TIMEOUT = Duration.ofSeconds(30);
+    private ServiceBuilder serviceBuilder;
+    private TableStore tableStore;
+    private StreamSegmentStore streamSegmentStore;
+    private PravegaConnectionListener serverListener = null;
+    private ConnectionPool connectionPool;
+    private TestingServer zkTestServer = null;
+    private ControllerWrapper controllerWrapper = null;
+    private Controller controller;
+    private ControllerServiceImpl controllerService;
+    private RequestTracker requestTracker = new RequestTracker(true);
+    private KeyValueTableFactory keyValueTableFactory;
+    private final int controllerPort = TestUtils.getAvailableListenPort();
+    private final String serviceHost = ENDPOINT;
+    private final int servicePort = TestUtils.getAvailableListenPort();
+    private final int containerCount = 4;
+    private final int startKeyRange = 10000;
+    private final int endKeyRange = 99999;
+    private final int valueSize = 10312;
+    private Random random = new Random(100);
+
+    @Before
+    public void setup() throws Exception {
```

Comment:
You can simplify the whole setup code significantly by using the `ClusterWrapper` instead. The `ClusterWrapper` encapsulates the server bootstrap logic and provides a simplified interface. See an example here: 

https://github.com/pravega/pravega/blob/22996fc928d762f886265a3dee75157fac944357/test/integration/src/test/java/io/pravega/test/integration/ClusterWrapperTest.java#L50

For your tests, you could disable Auth and TLS.